vortex-api/src/main/java/mil/army/usace/hec/vortex/geo/ZonalStatisticsCalculator.java

```diff
@@ -202,25 +202,93 @@ public static Map<String, Integer[]> createZoneMasks(Path pathToZoneDataset, Str
         return zoneMasks;
     }
 
-    private ZonalStatistics computeZonalStatistics(String id, Integer[] mask, float[] data) {
+    private ZonalStatistics computeZonalStatistics(String id, Integer[] mask, float[] data, List<String> statistics) {
 
         double noDataValue = grid.noDataValue();
 
+        List<Double> goodData = new ArrayList<>();
+
+        // compute average
         int count = 0;
         double sum = 0;
         for (int i = 0; i < data.length; i++) {
             double value = data[i];
             if (mask[i] == 1 && Double.compare(value, noDataValue) != 0) {
+
+                //if we've gotten to this point, it's a legit value; add it to the list of good data
+                goodData.add(value);
+
                 sum += value;
                 count++;
             }
         }
-
         double average = sum / count;
+        double min = Double.NaN;
+        double max = Double.NaN;
+        double median = Double.NaN;
+        double firstQuartile = Double.NaN;
+        double thirdQuartile = Double.NaN;
+        double q1;
+        double pctCellsGreaterThanZero = Double.NaN;
+        double pctCellsGreaterThanFirstQuartile = Double.NaN;
+        int numCellsGreaterThanZero = 0;
+
+        // if there are less than 4 goodData values, don't compute additional statistics
+        if (goodData.size() >= 4) {
+
+            if (statistics.size() != 0) {
+                // if user selects at least one of the statistics on panel 3
+                Collections.sort(goodData);
+
+                if (statistics.toString().contains("Min")) {
+                    min = goodData.get(0);
+                }
+                if (statistics.toString().contains("Max")) {
+                    max = goodData.get(count - 1);
+                }
+                if (statistics.toString().contains("Median")) {
+                    median = goodData.get((count + 1) / 2);
+                }
+
+                q1 = goodData.get((count + 1) / 4);
```

Comment:
Instead of computing here and assigning it to `firstQuartile`, move this computation inside the `if` block.

Reply:
I wanted to compute q1 as a separate variable from firstQuartile because a user could select % of cells above the first quartile but NOT the first quartile as a statistic. I only want to assign a value to variable firstQuartile if the user selects that statistic.